radix-engine/src/kernel/kernel.rs

```diff
@@ -201,6 +206,42 @@ pub struct Kernel<
     callback: &'g mut M,
 }
 
+struct CloseSubstateHandler<'a, M: KernelCallbackObject> {
```

Comment:
Quite some changes in this file that's not ready for review. Will skip this file for now.